rosetta-ide/src/main/java/com/regnosys/rosetta/ide/quickfix/ICodeActionProvider.java

```diff
@@ -0,0 +1,14 @@
+package com.regnosys.rosetta.ide.quickfix;
+
+import java.util.List;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.xtext.ide.server.codeActions.ICodeActionService2.Options;
+
+public interface ICodeActionProvider {
+	List<CodeAction> getCodeActions(Options options);
+
+	List<TextEdit> sortImportsResolution(EObject object);
```

Comment:
Surprised to see this - it is again tightly coupled to the sorting fix, not really to generic enough to handle other code actions.

How about a method `CodeAction resolveCodeAction(CodeAction codeAction)`, and the implementation can then check whether the title is `Sort imports`, and do the logic there?